crates/oxc_linter/src/rules/typescript/consistent_indexed_object_style.rs

```diff
@@ -0,0 +1,583 @@
+                                            if let TSSignature::TSIndexSignature(_) = member {
+                                                ctx.diagnostic(
+                                                    ConsistentIndexedObjectStyleDiagnostic(
+                                                        "record",
+                                                        "index signature",
+                                                        idx.span,
+                                                    ),
+                                                );
+                                            }
+                                        }
+                                    }
+                                }
+                                TSType::TSTypeReference(tref) => {
+                                    if !self.is_index_signature {
+                                        match &tref.type_name {
+                                            TSTypeName::IdentifierReference(iden) => {
+                                                if iden.name != decl.id.name {
+                                                    ctx.diagnostic(
+                                                        ConsistentIndexedObjectStyleDiagnostic(
+                                                            "record",
+                                                            "index signature",
+                                                            idx.span,
+                                                        ),
+                                                    );
+                                                }
+                                            }
+                                            TSTypeName::QualifiedName(_) => ctx.diagnostic(
+                                                ConsistentIndexedObjectStyleDiagnostic(
+                                                    "record",
+                                                    "index signature",
+                                                    idx.span,
+                                                ),
+                                            ),
+                                        }
+                                    }
+                                }
+                                TSType::TSUnknownKeyword(_) | TSType::TSAnyKeyword(_) => {
+                                    ctx.diagnostic(ConsistentIndexedObjectStyleDiagnostic(
+                                        "record",
+                                        "index signature",
+                                        idx.span,
+                                    ));
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
+                }
+            }
+            AstKind::TSTypeAliasDeclaration(al) => match &al.type_annotation {
```

Comment:
Does it work if we replace line 120 - line 355 with visit `AstKind::TSTypeLiteral` directly?

Reply:
not sure i got this one, can you elaborate?

Reply:
Like this?

```rust
fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
      match node.kind() {
          // --snip--
          AstKind::TSTypeLiteral(lit) => {
              if self.is_index_signature {
                  return;
              } 
              if lit.members.len() != 1 {
                  return;
              }
              let member = &lit.members[0];

              let TSSignature::TSIndexSignature(idx) = member else {
                  return;
              }

              // --snip--
          }
          AstKind::TSTypeReference(refercene) => {
              if !self.is_index_signature {
                  return;
              }
              if let TSTypeName::IdentifierReference(ident) = &refercene.type_name {
                  if ident.name == "Record" {
                      // snip
                  }
              } 
          }
          _ => {}
      }
  }
```

Reply:
I think I got it. Would you please take another look? :)

(sorry it took me so long)